numpyro/contrib/tfp/distributions.py

```diff
@@ -149,6 +149,12 @@ def is_discrete(self):
         # XXX: this should cover most cases
         return self.support is None
 
+    def sample_with_intermediates(self, key, sample_shape=()):
+        return self.sample(seed=key, sample_shape=sample_shape), []
+
+    def icdf(self, q):
```

Comment:
Good idea! But implementing an interface for such approximation can be tricky. I guess we can make something like `TruncatedApproximateReparam` so that users can decide if they want that approximation.

Btw, it seems that the log probability of Gamma there should be
```
return -gammaln(k) - k * jnp.log(theta) + (k - 1.) * jnp.log(X) - X / theta
```

Reply:
Yes, it could be tricky. If you knew how to estimate the support of the RV, as is done for the gamma example using the interval `(0, mean + 20. * stddev)`, then you could do it quite simply. 

And thanks! Good catch!